### ISSUE TITLE
Disabling glGenTextures for bug compatibility

### ISSUE THREAD

**Issue by a1batross** [FWGS/xash3d-fwgs#1479]
Some mods rely on texture_t `gl_texturenum` which we use internally for texture IDs, so I think about disabling `glGenTextures` in bugcomp mode. 

Should it be separate bugcomp mode or just enable every bugcompat mechanism at once?

**Comment by mittorn** [comment]
It still should be force-enabled for core contexts, it will be broken anyway.
Also, maybe we need auto-disable vbo in bugcomp mode if it breaks mods like snow war (but need to test it on windows first)
But what about detecting custom renderers some way before disabling renderer changes? Can we analyze client library or wrap LoadLibrary to detect if it loads opengl32? Or even wrap client's glBindTexture calls to prevent breaking the renderer? Any more ideas?

30 октября 2023 г. 23:30:47 GMT+03:00, Alibek Omarov ***@***.***> пишет:

**Comment by a1batross** [comment]
@mittorn 

Of course, we always can get `glBindTexture` addr and rewrite first few bytes to jmp to our `glBindTexture` impl. We did that before on x86 and ARM.

But I would rather not wrap mods that behave correctly. And there is seemingly no way to know whether mod using `glBindTexture` work as intended.

Mod where I have found broken textures also doesn't really load with core context, it just crashes there for some reason. O_o
